localstack/services/sqs/models.py

```diff
@@ -1154,6 +1157,11 @@ def receive(
                 else:
                     self.inflight.add(message)
 
+        # finally, add all empty groups back to the queue since there are potentially no inflight messages
+        # to put them back
+        for group in empty_groups:
+            self.message_group_queue.put_nowait(group)
```

Comment:
Should we check if the group is empty here? If in the meantime messages were added, would the group be in the queue twice?

Reply:
You are right, that can happen at the moment. Then we should probably make that empty check + requeue atomic. It wouldn't do any harm I think on a semantic level (the queues are "emptied" sequentially, so a subsequent duplicate message group "should" be empty) but it's probably begging for trouble in the long run, also from a performance perspective